### PR TITLE
spec updates

### PR DESCRIPTION
**Pull request by arnetheduck**
* use repeat_hash from spec in randao
* add some logging to state processing
* export crypto string converters in spec insulation layer
* enable block signature verification
* ssz: add support for arrays, remove custom data type code
* correctly handle previous and new block
* add trivial block processing tests

### CODE REVIEW

**Comment by arnetheduck** [comment]
@yglukhov right, I thought I did that but lost the code somewhere :)

File: beacon_chain/spec/beaconstate.nim [hunk 1]
```diff
@@ -6,11 +6,145 @@
...
+                     randao_commitment: Eth2Digest): Uint24 =
+  ## Process a deposit from Ethereum 1.0.
+  let msg = hash_tree_root((pubkey, withdrawal_credentials, randao_commitment))
+  assert BLSVerify(
+    pubkey, msg, proof_of_possession,
+    get_domain(state.fork_data, state.slot, DOMAIN_DEPOSIT))
+
+  let validator_pubkeys = mapIt(state.validator_registry, it.pubkey)
+
+  if pubkey notin validator_pubkeys:
```
Comment (mratsim) [inline beacon_chain/spec/beaconstate.nim]: I'm really confused about where the aggregation is supposed to happen.

If everyone has to carry a BeaconState with separated Validator pubkey, there is no point in for looping to aggregate them, and then verifying the aggregate.

We can just for loop verify.

File: beacon_chain/spec/beaconstate.nim [hunk 2]
```diff
@@ -6,11 +6,145 @@
...
+                              index: Uint24,
+                              new_status: ValidatorStatusCodes) =
+  ##  Update the validator status with the given ``index`` to ``new_status``.
+  ## Handle other general accounting related to this status update.
+  if new_status == ACTIVE:
+      activate_validator(state, index)
+  if new_status == ACTIVE_PENDING_EXIT:
+      initiate_validator_exit(state, index)
+  if new_status in [EXITED_WITH_PENALTY, EXITED_WITHOUT_PENALTY]:
+      exit_validator(state, index, new_status)
```
Comment (mratsim) [inline beacon_chain/spec/beaconstate.nim]: `case` statement?

File: beacon_chain/spec/beaconstate.nim (hunk 1 above)
Comment (mratsim) [reply beacon_chain/spec/beaconstate.nim]: Raised upstream https://github.com/ethereum/eth2.0-specs/issues/308

**Pull request merged**